### PR TITLE
DATA: Create `struct` for Model Name

### LINKED ISSUE #60: DATA: Create `struct` for Model Name

### PR DESCRIPTION
**Pull request by diogod3**
Closes #60 

I had the same idea as [@jodendaal ](https://github.com/hassanhabib/OpenAI.NET/issues/60#issuecomment-1474763194) for this, I used his suggestion with the `TypeConverter` for the serialization.

### CODE REVIEW

**Comment by hassanhabib** [comment]
@diogod3 I think I'm okay if you just created a struct with some hard defined values for now. Remove everything else.

**Comment by diogod3** [comment]
Sure, no problem. But how will users discover the `struct`?

**Comment by diogod3** [comment]
@BrianLParker I took a look at the endpoint, and it seems it returns all models available (I suppose it will list private models as well), but I couldn't understand how to check where the model should be used (text completion, code completion, chat completion, etc.).

This is an example of the data returned for each model (it returned 68 models):
```
 {
            "id": "babbage",
            "object": "model",
            "created": 1649358449,
            "owned_by": "openai",
            "permission": [
                {
                    "id": "modelperm-49FUp5v084tBB49tC4z8LPH5",
                    "object": "model_permission",
                    "created": 1669085501,
                    "allow_create_engine": false,
                    "allow_sampling": true,
                    "allow_logprobs": true,
                    "allow_search_indices": false,
                    "allow_view": true,
                    "allow_fine_tuning": false,
                    "organization": "*",
                    "group": null,
                    "is_blocking": false
                }
            ],
            "root": "babbage",
            "parent": null
        },
        {
            "id": "gpt-3.5-turbo",
            "object": "model",
            "created": 1677610602,
            "owned_by": "openai",
            "permission": [
                {
                    "id": "modelperm-ReelK0pSicCTFxpXe1JAOz29",
                    "object": "model_permission",
                    "created": 1679592889,
                    "allow_create_engine": false,
                    "allow_sampling": true,
                    "allow_logprobs": true,
                    "allow_search_indices": false,
                    "allow_view": true,
                    "allow_fine_tuning": false,
                    "organization": "*",
                    "group": null,
                    "is_blocking": false
                }
            ],
            "root": "gpt-3.5-turbo",
            "parent": null
        }
```

Nevertheless, I think it makes sense to offer a way within the lib to get a list of models and also to fetch and/or validate if a model exists.

**Comment by diogod3** [comment]
They have a model compatibility table on the documentation: https://platform.openai.com/docs/models/model-endpoint-compatibility

**Comment by BrianLParker** [comment]
@diogod3 @hassanhabib @hassanhabib
I cannot find the arrangement to identify what endpoint each model is for either. However, we will have a list of valid model names to be used for validation. If it is not the right model for the endpoint it will be caught by a denial/invalid request error.

File: Standard.AI.OpenAI/Models/Services/Foundations/Completions/ModelName.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿// --------------------------------------------------------------- 
+// Copyright (c) Coalition of the Good-Hearted Engineers 
+// ---------------------------------------------------------------
+
+namespace Standard.AI.OpenAI.Models.Services.Foundations.Completions
+{
+    public readonly struct ModelName
+    {
+        public static string Ada => "ada";
```
Comment (mabroukmahdhi) [inline Standard.AI.OpenAI/Models/Services/Foundations/Completions/ModelName.cs:9]: hey @diogod3 ,
why a static property here?

In general we use  `static property` when:
- The value of the constant needs to be calculated or accessed dynamically at runtime.
- The constant needs to be initialized using a complex operation or requires other code to be executed.
- The constant is only used within the class where it's declared.

Comment (diogod3) [reply Standard.AI.OpenAI/Models/Services/Foundations/Completions/ModelName.cs:9]: Oh, my bad, I'm sorry.

I refactored from the first version where the properties were structs also and forgot to change from static to const.

@mabroukmahdhi thanks, I'll get right on it.

**Comment by ElbekDeveloper** [comment]
@hassanhabib  do you think we can close this PR since our strategy on providing model names has changed.

**Comment by hassanhabib** [comment]
@ElbekDeveloper yea

**Pull request closed without merging**